detector/cve/cve20258088/cve20258088.go

```diff
@@ -0,0 +1,331 @@
+func (d Detector) Scan(ctx context.Context, scanRoot *scalibrfs.ScanRoot, px *packageindex.PackageIndex) (inventory.Finding, error) {
+	var findings []*inventory.PackageVuln
+	// Match any PE file that could be WinRAR, RAR, UnRAR, UnRAR.dll, including renamed and portable versions
+	peRegex := regexp.MustCompile(`(?i)(winrar|rar|unrar).*?(\.exe|\.dll)$`)
+
+	// === Phase 1: Installed packages via package index ===
+	if px != nil {
+		for _, pkg := range px.GetAll() {
+			nameLower := strings.ToLower(pkg.Name)
+			if strings.Contains(nameLower, "winrar") {
+				normalizedVersion := normalizeVersion(pkg.Version)
+
+				if isAffectedVersion(normalizedVersion) {
+					log.Infof("Vulnerable WinRAR package installation found(Uninstall them): Package %s, Version: %s", pkg.Name, normalizedVersion)
+
+					affected := []osvschema.Affected{{
+						Package: osvschema.Package{
+							Name:      pkg.Name,
+							Ecosystem: "generic",
+						},
+						Ranges: []osvschema.Range{{
+							Type: "SEMVER",
+							Events: []osvschema.Event{{
+								Introduced: "0",
+							}, {
+								Fixed: "7.13",
+							}},
+						}},
+						Versions: []string{normalizedVersion},
+					}}
+
+					vuln := &inventory.PackageVuln{
+						Package: pkg,
+						Vulnerability: osvschema.Vulnerability{
+							ID:       "CVE-2025-8088",
+							Summary:  "WinRAR vulnerable version detected (installed package)",
+							Details:  fmt.Sprintf("Installed WinRAR %s detected via package index", normalizedVersion),
+							Affected: affected,
+						},
+					}
+
+					findings = append(findings, vuln)
+				}
+			}
+		}
+	}
+
+	// === Phase 2: Filesystem scan (existing logic) ===
+	err := fs.WalkDir(scanRoot.FS, scanRoot.Path, func(path string, d fs.DirEntry, err error) error {
+		if err != nil || d.IsDir() {
+			log.Debugf("cve20258088: Skipping directory or directory entry: %s : error %s", path, err)
+			return nil
+		}
+
+		ext := strings.ToLower(filepath.Ext(d.Name()))
+		if ext != ".exe" && ext != ".dll" {
+			return nil
+		}
+
+		confidence := 0
+		var component string
+
+		// Indicator 1: Filename match
+		if peRegex.MatchString(d.Name()) {
+			confidence += 2
+			matches := peRegex.FindStringSubmatch(d.Name())
+			if len(matches) > 1 {
+				component = matches[1]
+			}
+		}
+
+		// Indicator 2: PE resource analysis - extract version info from PE resources
+		// Now check PE resources for ALL .exe/.dll files, not just regex matches
+		version, prodName := extractPEVersion(path)
+		if prodName != "" {
+			// Check if the product name indicates WinRAR/RAR/UnRAR
+			prodNameLower := strings.ToLower(prodName)
+			if strings.Contains(prodNameLower, "winrar") || strings.Contains(prodNameLower, "rar") || strings.Contains(prodNameLower, "unrar") {
+				confidence += 2
+				if component == "" {
+					component = prodName
+				}
+			}
+		}
+
+		// Skip if we have no indication this is a WinRAR-related file
+		if confidence == 0 {
```

Comment:
Is this branch needed? AFAICS we only store anything if confidence>=2 lower down